Feedback word level feedback pr

dcsiebe opened the pull request:
#73 Should be merged first.

dcsiebe committed:
docs(feedback): re-sending of header on resume

Note that it is required to re-send the audio header on resume as it is
quite important.

dcsiebe committed:
feat(feedback): word level feedback

Add feedback on words when feedback of a sentence is given.

kvdb commented on docs/websocket/feedback.md:
```diff
@@ -123,6 +143,16 @@ id   | `string` | **Required** The id of the feedback to pause.
 again processed by the server.
+Currently **it is required to re-send the audio header when resuming
+feedback**. Failing to do so will fail the feedback as the server can't
+recognise the audio format without a header.
```
I thought the server could be intelligent about the header being absent or not.
Well, if the SDK takes care of it, it's fine either way though.

dcsiebe replied:
I think this feedback belongs in #73 :wink:

joggienl replied:
it is marked as added in this PR so the feedback does belong here.
The server can't do that now, so the SDK will do it and the docs document you should do it (if you're not using the SDK).
So its fine for now indeed.

dcsiebe replied:
It's only marked as added in this PR as #73 isn't merged yet. I added a comment with this PR stating "#73 Should be merged first.", I'm not saying this without reason, if you decide to ignore that comment useless discussions like this can arise.

joggienl replied:
I think you should have merged the two PR's to one to get review on.

dcsiebe replied:
Solved in #73.

kvdb commented on docs/websocket/feedback.md:
```diff
@@ -63,14 +63,30 @@ the following json is sent as a progressive result:
+    {
+      "expected": "gif",
+      "recognized": "jif",
+      "correct": false
```
This is ok, I'm just wondering if this covers all situation, I don't think so.
AFAIK, these are possibilities: COR: correct, DEL: deletion, INS: insersion, SUB: substitution

kvdb replied:
Let's keep this PR open for now, until we've tested this a bit more.

dcsiebe replied:
Correct: 
```
{
  "expected": "spam",
  "recognized": "spam",
  "correct": true
}
```

Deletion:
```
{
  "expected": "spam",
  "recognized": "",
  "correct": false
}
```

Insertion:
```
{
  "expected": "",
  "recognized": "spam",
  "correct": false
}
```

Substitution:
```
{
  "expected": "spam",
  "recognized": "eggs",
  "correct": false
}
```

As discussed in this mornings standup I'll add an extra field to explicitly state the error.

dcsiebe committed:
feat(feedback): error field for word

Add an extra field to show what kind of error was detected.

kvdb commented:
+1

joggienl commented on docs/websocket/feedback.md:
```diff
@@ -123,6 +159,16 @@ id   | `string` | **Required** The id of the feedback to pause.
+recognise the audio format without a header.
+
+!!! note
+    It appears to be valid to have a wave file with multiple headers.
```
Did we test that? We could (re)generate the wave file without the extra headers btw.

dcsiebe replied:
It seems to work in the tests and I read something about this in a description of the wave format.

joggienl replied:
👍

joggienl commented on docs/websocket/feedback.md:
```diff
@@ -123,6 +159,16 @@ id   | `string` | **Required** The id of the feedback to pause.
+
+!!! warning
+    It is required to re-send the audio header upon resume. The audio
+    processing will fail if the header isn't re-sent.
```
Is this extra note necessary since you already put the message **it is required to re-send the audio header when resuming feedback** before?

dcsiebe replied:
Shouldn't be, but I thought it would be best to emphasize it since I recon people will be reading over it.

joggienl replied:
It is not necessary to have this redundant comment, please remove it.

joggienl commented on docs/websocket/feedback.md:
```diff
@@ -63,14 +63,46 @@ the following json is sent as a progressive result:
 ---------|--------|------------
 sentence | `int`  | The index of the sentence, starting at `0`.
 feedback | `bool` | Feedback score, `false` means bad and `true` means good.
+words    | `list` | The expected and recognized words in the sentence.
```
At this point I assume that we will always return ALL words that were expected in a single sentence. If so, it might be good to note that as wel.

dcsiebe replied:
Done

dcsiebe committed:
docs(feedback): all expected words are returned

More explicitly state that all expected words are returned in the
feedback.

dcsiebe committed:
Merge branch 'feedback_api' into feedback_word_level_feedback_pr

dcsiebe committed:
refactor(feedback): remove redundant warning

Remove warning about the required re-sending of the audio header as a
warning is already in the text.

kvdb commented:
Please close MR or fix conflict

joggienl committed:
Merge branch 'feedback_api' into feedback_word_level_feedback_pr
